packages/react-router/lib/router/utils.ts

```diff
@@ -425,9 +426,13 @@ type RequireOne<T, Key = keyof T> = Exclude<
  * related properties to a route
  */
 export interface LazyRouteFunction<R extends AgnosticRouteObject> {
-  (): Promise<RequireOne<Omit<R, ImmutableRouteKey>>>;
+  (): Promise<RequireOne<Omit<R, UnsupportedLazyRouteFunctionKey>>>;
 }
 
+export type LazyRouteFunctionCache<
```

Comment:
nit, but I might just inline this type into the `const` definition if it's not needed for reuse anywhere else